### ISSUE TITLE
loss: nan - val_loss nan

### ISSUE THREAD

**Issue by stsmall** [kr-colab/ReLERNN#26]
Hi,
For some of my training runs, not all, the loss values eventually start being reported as 'nan'.
 Epoch 308/1000
 99/100 [============================>.] - ETA: 2s - loss: 0.2445Epoch 1/1000
100/100 [==============================] - 328s 3s/step - loss: 0.2445 - val_loss: 0.2342
Epoch 309/1000
 99/100 [============================>.] - ETA: 3s - loss: nanEpoch 1/1000
100/100 [==============================] - 332s 3s/step - loss: nan - val_loss: nan
Epoch 310/1000
 99/100 [============================>.] - ETA: 2s - loss: nanEpoch 1/1000
100/100 [==============================] - 325s 3s/step - loss: nan - val_loss: nan

It usually terminates successfully after a repeat of 'nan'. Just wanted to verify that there is nothing wrong with the training that would influence the predictions in this case.
thanks,
@stsmall

**Comment by jradrion** [comment]
Hi @stsmall,

While it isn't terribly common, we have seen this behavior before. We're not entirely sure what is causing it or how to fix it. That being said, the model weights are updated as training progresses, so training terminating with `nan` will still allow for valid predictions, provided that previous epochs completed without loss = `nan`.  In the example you provided, the weights used for prediction will be taken from the training epoch with the smallest lost prior to epoch 309.

One thing I will add to my to-do list is terminating training immediately upon a `nan`, to avoid additional epochs (i.e. wasted time). I'll see if I can implement this soon. 

Best,
Jeff

**Comment by stsmall** [comment]
Hi Jeff,
Thanks! It usually ended up terminating after 10-30 more nan.
thanks,
@stsmall

**Comment by jradrion** [comment]
I added code to terminate training upon `loss = Nan`.
